### ISSUE TITLE
MaxDeliveryCount not respected

### ISSUE THREAD

**Issue by jnsvd** [PejmanNik/lazvard-message#13]
Messages are delivered infinitely many times on subscriptions when an error occurs during processing. Only when the message can't be delivered (after I've shutdown the processor), the message will be dead lettered, after which it shows an error like:

```
the message 1-62130720 in subscription request-processor has reached the maximum delivery count 3 and will be dead-lettered
```

but earlier a higher delivery count was logged:

```
delivered message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with DeliveryTag 70, 56, 232, 35, 141, 247, 76, 71, 143, 245, 25, 150, 54, 136, 116, 44, DeliveryCount 12
```

Configuration:

```toml
[Server]
# IP to listen on
IP = "localhost"
# Port to listen on
Port = 5671
# use Https with a valid certificate, default is false
UseHttps = false
# Use built-in certificate manager (dotnet dev-certs) instead of CertificatePath
UseBuiltInCertificateManager = true
# The path to trusted X.509 certificate (PFX - PKCS #12), required only when UseBuiltInCertificateManager is false
CertificatePath = ""
# The certificate password, optional
CertificatePassword = ""

[AMQP]
# Connection idle timeout in milliseconds for AMQP connection, default is 4 minutes
ConnectionIdleTimeOut = 240000
# Max AMQP frame size in byte, default is 64Kib
MaxFrameSize = 65536
# Max AMQP message size in byte, default is 64MiB
MaxMessageSize = 67108864

[[Topics]]
# Topic name
Name = "requests"

[[Topics.Subscriptions]]
# Subscription name
Name = "request-processor"
# Number of maximum deliveries. The default value is 50 times
MaxDeliveryCount = 3
# ISO 8061 lock duration for the subscription. The default value is 1 minute
LockDuration = "PT2M"
```

Full logs:

```
  _                 _                              _
 | |       __ _    (_) __   __   __ _   _ __    __| |
 | |      / _` |   | | \ \ / /  / _` | | '__|  / _` |
 | |___  | (_| |   | |  \ V /  | (_| | | |    | (_| |
 |_____|  \__,_|  _/ |   \_/    \__,_| |_|     \__,_|
                 |__/

  ServiceBus Simulation
────────────────────────────────────────────────────────────────────────────────

Lajvard ServiceBus service is successfully listening at http://localhost:5671

╔══════════════════════════════════════════════════════════════════════════════╗
║                                                                              ║
║ ConnectionString:                                                            ║
║ Endpoint=sb://localhost:5671/;SharedAccessKeyName=RootManageSharedAccessKey; ║
║ SharedAccessKey=1;UseDevelopmentEmulator=true;                               ║
║                                                                              ║
╚══════════════════════════════════════════════════════════════════════════════╝


trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor/$management
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor/$deadletterqueue
trce: Lazvard.Message.Amqp.Server.Broker[0]
      accept new connecting from (null)
trce: Lazvard.Message.Amqp.Server.Broker[0]
      open connecting 2
trce: Lazvard.Message.Amqp.Server.SubscriptionHandler[0]
      attach sending link from source /requests/Subscriptions/request-processor and link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with Identifier 7
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 1 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 20 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 20 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 20 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 20 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 20 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 20 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 20 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 20 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 20 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      write message 1 to subscription request-processor channel
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      process message 1-62130720 in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      trying to deliver message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.MessageQueue[0]
      add lock f975395d-7c13-4e34-afb5-f43dd9aa4bd4 until 638708128740281621 for message 1
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      delivered message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with DeliveryTag 93, 57, 117, 249, 19, 124, 52, 78, 175, 181, 244, 61, 217, 170, 75, 212, DeliveryCount 0
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      delivering message 1-62130720 in subscription request-processor to consumer  was Successful
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition f975395d-7c13-4e34-afb5-f43dd9aa4bd4 with status amqp:modified:list
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition f975395d-7c13-4e34-afb5-f43dd9aa4bd4 was successful
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      process message 1-62130720 in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      trying to deliver message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.MessageQueue[0]
      add lock edf6eb9e-dcf9-459b-bfdd-632e71892fe3 until 638708128775375417 for message 1
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      delivered message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with DeliveryTag 158, 235, 246, 237, 249, 220, 155, 69, 191, 221, 99, 46, 113, 137, 47, 227, DeliveryCount 1
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      delivering message 1-62130720 in subscription request-processor to consumer  was Successful
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition edf6eb9e-dcf9-459b-bfdd-632e71892fe3 with status amqp:modified:list
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition edf6eb9e-dcf9-459b-bfdd-632e71892fe3 was successful
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      process message 1-62130720 in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      trying to deliver message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.MessageQueue[0]
      add lock 0cb146b9-b7b4-4bb8-8bbc-d564ac684a87 until 638708128809896317 for message 1
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      delivered message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with DeliveryTag 185, 70, 177, 12, 180, 183, 184, 75, 139, 188, 213, 100, 172, 104, 74, 135, DeliveryCount 2
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      delivering message 1-62130720 in subscription request-processor to consumer  was Successful
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition 0cb146b9-b7b4-4bb8-8bbc-d564ac684a87 with status amqp:modified:list
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition 0cb146b9-b7b4-4bb8-8bbc-d564ac684a87 was successful
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      process message 1-62130720 in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      trying to deliver message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.MessageQueue[0]
      add lock 7ec02be7-422b-46a2-9815-8bc7fcd4cc53 until 638708128842826295 for message 1
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      delivered message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with DeliveryTag 231, 43, 192, 126, 43, 66, 162, 70, 152, 21, 139, 199, 252, 212, 204, 83, DeliveryCount 3
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      delivering message 1-62130720 in subscription request-processor to consumer  was Successful
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition 7ec02be7-422b-46a2-9815-8bc7fcd4cc53 with status amqp:modified:list
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition 7ec02be7-422b-46a2-9815-8bc7fcd4cc53 was successful
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      process message 1-62130720 in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      trying to deliver message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.MessageQueue[0]
      add lock a87cf0f8-4acc-4d51-880a-fa81a82ca4b2 until 638708128875608239 for message 1
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      delivered message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with DeliveryTag 248, 240, 124, 168, 204, 74, 81, 77, 136, 10, 250, 129, 168, 44, 164, 178, DeliveryCount 4
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      delivering message 1-62130720 in subscription request-processor to consumer  was Successful
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition a87cf0f8-4acc-4d51-880a-fa81a82ca4b2 with status amqp:modified:list
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition a87cf0f8-4acc-4d51-880a-fa81a82ca4b2 was successful
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      process message 1-62130720 in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      trying to deliver message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.MessageQueue[0]
      add lock 6ec12711-f130-4063-bf08-793c32737d0a until 638708128907859490 for message 1
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      delivered message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with DeliveryTag 17, 39, 193, 110, 48, 241, 99, 64, 191, 8, 121, 60, 50, 115, 125, 10, DeliveryCount 5
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      delivering message 1-62130720 in subscription request-processor to consumer  was Successful
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition 6ec12711-f130-4063-bf08-793c32737d0a with status amqp:modified:list
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      process message 1-62130720 in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      trying to deliver message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition 6ec12711-f130-4063-bf08-793c32737d0a was successful
trce: Lazvard.Message.Amqp.Server.MessageQueue[0]
      add lock b330f8bf-2c76-4a40-a6a5-3cc24bc7af08 until 638708128951425993 for message 1
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      delivered message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with DeliveryTag 191, 248, 48, 179, 118, 44, 64, 74, 166, 165, 60, 194, 75, 199, 175, 8, DeliveryCount 6
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      delivering message 1-62130720 in subscription request-processor to consumer  was Successful
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition b330f8bf-2c76-4a40-a6a5-3cc24bc7af08 with status amqp:modified:list
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition b330f8bf-2c76-4a40-a6a5-3cc24bc7af08 was successful
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      process message 1-62130720 in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      trying to deliver message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.MessageQueue[0]
      add lock aeaa1c22-3b39-4526-95d3-b81cbb4b80cd until 638708128983970469 for message 1
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      delivered message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with DeliveryTag 34, 28, 170, 174, 57, 59, 38, 69, 149, 211, 184, 28, 187, 75, 128, 205, DeliveryCount 7
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      delivering message 1-62130720 in subscription request-processor to consumer  was Successful
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition aeaa1c22-3b39-4526-95d3-b81cbb4b80cd with status amqp:modified:list
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition aeaa1c22-3b39-4526-95d3-b81cbb4b80cd was successful
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      process message 1-62130720 in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      trying to deliver message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.MessageQueue[0]
      add lock e6249d73-dfd5-437b-b19a-9cb19559be2d until 638708129016404161 for message 1
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      delivered message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with DeliveryTag 115, 157, 36, 230, 213, 223, 123, 67, 177, 154, 156, 177, 149, 89, 190, 45, DeliveryCount 8
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      delivering message 1-62130720 in subscription request-processor to consumer  was Successful
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition e6249d73-dfd5-437b-b19a-9cb19559be2d with status amqp:modified:list
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition e6249d73-dfd5-437b-b19a-9cb19559be2d was successful
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      process message 1-62130720 in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      trying to deliver message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.MessageQueue[0]
      add lock 4a3cc6ed-ad5e-404f-9c31-474695f5494f until 638708129050099286 for message 1
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      delivered message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with DeliveryTag 237, 198, 60, 74, 94, 173, 79, 64, 156, 49, 71, 70, 149, 245, 73, 79, DeliveryCount 9
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      delivering message 1-62130720 in subscription request-processor to consumer  was Successful
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition 4a3cc6ed-ad5e-404f-9c31-474695f5494f with status amqp:modified:list
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition 4a3cc6ed-ad5e-404f-9c31-474695f5494f was successful
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      process message 1-62130720 in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      trying to deliver message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.MessageQueue[0]
      add lock c58e5234-f614-4018-813e-666051aaa0cd until 638708129082265780 for message 1
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      delivered message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with DeliveryTag 52, 82, 142, 197, 20, 246, 24, 64, 129, 62, 102, 96, 81, 170, 160, 205, DeliveryCount 10
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      delivering message 1-62130720 in subscription request-processor to consumer  was Successful
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition c58e5234-f614-4018-813e-666051aaa0cd with status amqp:modified:list
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition c58e5234-f614-4018-813e-666051aaa0cd was successful
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      process message 1-62130720 in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      trying to deliver message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.MessageQueue[0]
      add lock cf306b54-a2bc-47bc-9d04-6a58bd79c4c2 until 638708129103564325 for message 1
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      delivered message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with DeliveryTag 84, 107, 48, 207, 188, 162, 188, 71, 157, 4, 106, 88, 189, 121, 196, 194, DeliveryCount 11
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      delivering message 1-62130720 in subscription request-processor to consumer  was Successful
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition cf306b54-a2bc-47bc-9d04-6a58bd79c4c2 with status amqp:modified:list
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition cf306b54-a2bc-47bc-9d04-6a58bd79c4c2 was successful
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      process message 1-62130720 in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      trying to deliver message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.MessageQueue[0]
      add lock 23e83846-f78d-474c-8ff5-19963688742c until 638708129107359965 for message 1
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      delivered message 1 to amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[]) with DeliveryTag 70, 56, 232, 35, 141, 247, 76, 71, 143, 245, 25, 150, 54, 136, 116, 44, DeliveryCount 12
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      delivering message 1-62130720 in subscription request-processor to consumer  was Successful
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition 23e83846-f78d-474c-8ff5-19963688742c with status amqp:released:list
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process delivery tag in disposition 23e83846-f78d-474c-8ff5-19963688742c was failed
trce: Lazvard.Message.Amqp.Server.Consumer[0]
      process credit 0 for link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      close link amqp://localhost/-2ee79f55;0:5:6:source(address:/requests/Subscriptions/request-processor,filter:[])
trce: Lazvard.Message.Amqp.Server.Broker[0]
      close connecting 2
trce: Lazvard.Message.Amqp.Server.MessageQueue[0]
      message 1 with lock 23e83846-f78d-474c-8ff5-19963688742c expired and requeued
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      no consumers to deliver the message 1-62130720 in subscription request-processor, waiting...
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      waiting to receive a message in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      process message 1-62130720 in subscription request-processor
fail: Lazvard.Message.Amqp.Server.Subscription[0]
      the message 1-62130720 in subscription request-processor has reached the maximum delivery count 3 and will be dead-lettered
fail: Lazvard.Message.Amqp.Server.Subscription[0]
      can not re-enqueue message 1-62130720 in subscription request-processor
trce: Lazvard.Message.Amqp.Server.Subscription[0]
      no consumers to deliver the message 1-64841458 in subscription request-processor/$deadletterqueue, waiting...
```

**Comment by PejmanNik** [comment]
Hi, Thank you for reporting this. it should be fixed in the version 0.6.0 https://github.com/PejmanNik/lazvard-message/releases/tag/v0.6.0

**Comment by jnsvd** [comment]
I've tried with the new version and confirm that it works as expected now. Thanks for your quick fix and for providing and maintaining the simulator!
